File: software/examples/vbnet/ExampleThreshold.vb
Imports System
Imports Tinkerforge

Module ExampleThreshold
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Industrial Dual 0-20mA Bricklet 2.0

    ' Callback subroutine for voltage callback
    Sub CurrentCB(ByVal sender As BrickletIndustrialDual020mAV2, ByVal channel As Byte, _
                  ByVal current As Integer)
        Console.WriteLine("Channel: " + channel.ToString())
        Console.WriteLine("Current: " + (current/1000000.0).ToString() + " mA")
        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim id020 As New BrickletIndustrialDual020mAV2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register voltage callback to subroutine CurrentCB
        AddHandler id020.CurrentCallback, AddressOf CurrentCB

        ' Configure threshold for current "greater than 10 mA"
        ' with a debounce period of 10s (10000ms)
        id020.SetCurrentCallbackConfiguration(0, 10000, False, ">"C, 10*1000000, 0)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/industrial_dual_0_20ma_v2_bricklet"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Industrial Dual 0-20mA Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	id020, _ := industrial_dual_0_20ma_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	id020.RegisterCurrentCallback(func(channel uint8, current int32) {
		fmt.Printf("Channel: %d\n", channel)
		fmt.Printf("Current: %f mA\n", float64(current)/1000000.0)
		fmt.Println()
	})

	// Configure threshold for current (channel 0) "greater than 10 mA"
	// with a debounce period of 10s (10000ms).
	id020.SetCurrentCallbackConfiguration(0, 10000, false, '>', 10*1000000, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
